wsclient/src/test/java/org/hpccsystems/ws/client/WSECLTests.java

```diff
@@ -0,0 +1,181 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+package org.hpccsystems.ws.client;
+
+import static org.junit.Assume.assumeTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.List;
+
+import org.hpccsystems.ws.client.utils.Connection;
+import org.hpccsystems.ws.client.wrappers.gen.wsworkunits.WUPublishWorkunitResponseWrapper;
+import org.hpccsystems.ws.client.wrappers.wsworkunits.QueryResultWrapper;
+import org.hpccsystems.ws.client.wrappers.wsworkunits.WorkunitWrapper;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+public class WSECLTests extends BaseRemoteTest
+{
+    private static HPCCWsWorkUnitsClient wswuclient = null;
+    private static boolean hasPublishedQuery = false;
+    private final static String wsECLPort = System.getProperty("wseclport", "8002");
+    private final static String eclScriptName = "SimpleFunction.ecl";
+    private static Connection wseclConn = null;
+
+    @BeforeClass
+    public static void setup() throws Exception
+    {
+        wswuclient = wsclient.getWsWorkunitsClient(); //for publishing queries
```

Comment:
this method runs before the tests.
Here we try to publish an ecl query, then confirm it was actually published, and setup the wsecl connection